### PR TITLE
Add traceid to decision logs

### PR DESCRIPTION
**Pull request by be-a-bee**
https://github.com/open-policy-agent/opa/issues/5230

thank you for your patience @ashutosh-narkar  . kindly review

### CODE REVIEW

File: test/e2e/distributedtracing/distributedtracing_test.go [hunk 1]
```diff
@@ -147,5 +148,52 @@ func TestServerSpan(t *testing.T) {
 		if got, expected := spans[0].Parent.SpanID(), parentSpanID; got != expected {
 			t.Errorf("expected span to be child of %v, got parent %v", expected, got)
 		}
+
+		var entry test.LogEntry
+		var found bool
+
+		for _, entry = range consoleLogger.Entries() {
+			t.Log(entry.Message)
```
Comment (ashutosh-narkar) [inline test/e2e/distributedtracing/distributedtracing_test.go:156]: Nit: We probably don't need to log the entry.

File: test/e2e/distributedtracing/distributedtracing_test.go [hunk 2]
```diff
@@ -88,7 +92,7 @@ func TestMain(m *testing.M) {
 		resp.body.count == 1
 	}`
 	module := fmt.Sprintf(moduleFmt, ts.URL)
-	pluginsManager, err := e2e.TestAuthzServerWithWithOpts(module, "envoy/authz/allow", ":9191", plugins.WithTracerProvider(tracerProvider))
+	pluginsManager, err := e2e.TestAuthzServerWithWithOpts(module, "envoy/authz/allow", ":9191", plugins.WithTracerProvider(tracerProvider), plugins.ConsoleLogger(consoleLogger))
```
Comment (ashutosh-narkar) [inline test/e2e/distributedtracing/distributedtracing_test.go:95]: Do you intend to use the console logger somewhere in this test?

File: test/e2e/testing.go [hunk 3]
```diff
@@ -37,6 +57,22 @@ func TestAuthzServerWithWithOpts(module string, path string, addr string, opts .
 		return nil, err
 	}
 	m.Register(plugin.PluginName, internal.New(m, cfg))
+
+	//services := []string{"s1", "s3"}
+
+	m.Register("test_plugin", &testPlugin{})
```
Comment (ashutosh-narkar) [inline test/e2e/testing.go:63]: Can you please help me understand why we're registering this and how we're using it?

**Review by ashutosh-narkar — COMMENTED** [review]
@LionOnTheChase the changes look good. Some comments inline. Thanks!

**Comment by ashutosh-narkar** [comment]
@LionOnTheChase can you please sign your commit. Thanks!

File: test/e2e/distributedtracing/distributedtracing_test.go (hunk 2 above)
Comment (LionOnTheChase) [reply test/e2e/distributedtracing/distributedtracing_test.go:95]: @ashutosh-narkar  , yes. 

I'm using it to capture the log entries, identify a decision log in them, and then assert for the presence of trace_id and span_id in the decision log. 

see line 155 to 159

File: test/e2e/testing.go (hunk 3 above)
Comment (LionOnTheChase) [reply test/e2e/testing.go:63]: I needed to enable decision_logs .  

Looking at https://github.com/open-policy-agent/opa-envoy-plugin/blob/064d6406c230b16ff17fd8c03500b42a89ec5914/envoyauth/evaluation_test.go#L212 , I figured that this is one way we could do it. 

Please suggest if there is any better way.

**Comment by ashutosh-narkar** [comment]
Closing in favor of https://github.com/open-policy-agent/opa-envoy-plugin/pull/470.

**Pull request closed without merging**